crates/red_knot_python_semantic/resources/mdtest/narrow/match.md

```diff
@@ -0,0 +1,17 @@
+# Narrowing for `match` statements
+
+## Single `match` pattern
+
+```py
+x = None if flag else 1
+reveal_type(x)  # revealed: None | Literal[1]
+
+y = 0
+
+match x:
+    case None:
+        y = x
+
+# TODO intersection simplification: should be just Literal[0] | None
+reveal_type(y)  # revealed: Literal[0] | None | Literal[1] & None
+```
```

Comment:
Copied over from [this comment](https://github.com/astral-sh/ruff/pull/13719#discussion_r1800188105) by @carljm